aptos-move/framework/aptos-framework/sources/Genesis.move

```diff
@@ -101,93 +108,62 @@ module AptosFramework::Genesis {
         // this needs to be called at the very end
         ChainId::initialize(core_resource_account, chain_id);
         Reconfiguration::initialize(core_resource_account);
-        Block::initialize_block_metadata(core_resource_account);
+        Block::initialize_block_metadata(core_resource_account, epoch_interval);
         Timestamp::set_time_has_started(core_resource_account);
     }
 
     /// Sets up the initial validator set for the network.
-    /// The validator "owner" accounts, their UTF-8 names, and their authentication
-    /// keys are encoded in the `owners`, `owner_names`, and `owner_auth_key` vectors.
+    /// The validator "owner" accounts, and their authentication
+    /// keys are encoded in the `owners` and `owner_auth_key` vectors.
     /// Each validator signs consensus messages with the private key corresponding to the Ed25519
     /// public key in `consensus_pubkeys`.
-    /// Each validator owner has its operation delegated to an "operator" (which may be
-    /// the owner). The operators, their names, and their authentication keys are encoded
-    /// in the `operators`, `operator_names`, and `operator_auth_keys` vectors.
     /// Finally, each validator must specify the network address
     /// (see types/src/network_address/mod.rs) for itself and its full nodes.
-    fun create_initialize_owners_operators(
+    fun create_initialize_validators(
         core_resource_account: signer,
-        owners: vector<signer>,
-        owner_names: vector<vector<u8>>,
+        owners: vector<address>,
         owner_auth_keys: vector<vector<u8>>,
         consensus_pubkeys: vector<vector<u8>>,
-        operators: vector<signer>,
-        operator_names: vector<vector<u8>>,
-        operator_auth_keys: vector<vector<u8>>,
         validator_network_addresses: vector<vector<u8>>,
         full_node_network_addresses: vector<vector<u8>>,
+        staking_distribution: vector<u64>,
     ) {
```

Comment:
is this legacy ugliness because move (used to) not support passing in a vec<validatorconfigs> or even just taking the vec<addrs> and pulling from there, this is .. yuck